linkml/generators/pydanticgen/template.py

```diff
@@ -499,8 +535,58 @@ def __iter__(self) -> Generator[Import, None, None]:
         for i in self.imports:
             yield i
 
-    def __getitem__(self, item: int) -> Import:
-        return self.imports[item]
+    def __getitem__(self, item: Union[int, str]) -> Import:
+        if isinstance(item, int):
+            return self.imports[item]
+        elif isinstance(item, str):
+            # the name of the module
+            an_import = [i for i in self.imports if i.module == item]
+            if len(an_import) == 0:
+                raise KeyError(f"No import with module {item} was found.\nWe have: {self.imports}")
+            return an_import[0]
+        else:
+            raise TypeError("Can only index with an int or a string as the name of the module," "\nGot: {type(item)}")
```

Comment:
Oop missing an f string

Reply:
done in https://github.com/linkml/linkml/pull/2208/commits/7fed49b3764d3f8c1ee84422444e675d0baec99c